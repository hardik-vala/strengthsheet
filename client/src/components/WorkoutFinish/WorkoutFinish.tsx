import { View } from "react-native";
import { Appbar, Avatar, Text } from "react-native-paper";
import { WorkoutTemplate } from "../../models/Workout/WorkoutTemplate";

interface WorkoutFinishProps {
  workoutTemplate: WorkoutTemplate;
  onDismiss: () => void;
}

export function WorkoutFinish({
  workoutTemplate,
  onDismiss,
}: WorkoutFinishProps) {
  return (
    <View>
      <Appbar.Header statusBarHeight={15}>
        <Appbar.Action icon="close" onPress={onDismiss} />
      </Appbar.Header>
      <View style={{ alignItems: "center", marginTop: 15 }}>
        <Text
          variant="headlineSmall"
          style={{ fontWeight: "bold", textAlign: "center" }}
        >{`${workoutTemplate.displayName} workout complete!`}</Text>
        <Avatar.Icon
          color="green"
          icon="check-circle"
          size={80}
          style={{ marginTop: 5 }}
        />
      </View>
    </View>
  );
}
