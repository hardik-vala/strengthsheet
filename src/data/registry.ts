import { Circuit, Exercise, ExerciseUnit, SetType } from "../models/Workout/Core";
import { WorkoutTemplate } from "../models/Workout/WorkoutTemplate";

export type WorkoutTemplateRegistry = { [k: string]: WorkoutTemplate };

export const EXERCISE_REGISTRY: { [k: string]: Exercise } = {
  hamstring_curl_machine: {
    key: "hamstring_curl_machine",
    displayName: "Hamstring Curl (Machine)",
    measures: [
      { key: "lb", displayName: "lb", unit: ExerciseUnit.lb },
      { key: "reps", displayName: "reps", unit: ExerciseUnit.reps },
    ]
  },
  hip_thrust: {
    key: "hip_thrust",
    displayName: "Hip Thrust (Weighted)",
    measures: [
      { key: "lb", displayName: "lb", unit: ExerciseUnit.lb },
      { key: "reps", displayName: "reps", unit: ExerciseUnit.reps },
    ]
  },
  hip_thrust_single_leg: {
    key: "hip_thrust_single_leg",
    displayName: "Single-Leg Hip Thrust (Weighted)",
    measures: [
      { key: "lb", displayName: "lb", unit: ExerciseUnit.lb },
      { key: "reps", displayName: "reps", unit: ExerciseUnit.reps },
    ]
  },
  rowing_machine: {
    key: "rowing_machine",
    displayName: "Rowing (Machine)",
    measures: [
      { key: "meters", displayName: "meters", unit: ExerciseUnit.meters },
      { key: "time", displayName: "time", unit: ExerciseUnit.time }
    ],
  },
  squat_split_bulgarian_bodyweight: {
    key: "squat_split_bulgarian_bodyweight",
    displayName: "Bulgarian Split Squat (Bodyweight)",
    measures: [
      { key: "reps", displayName: "reps", unit: ExerciseUnit.reps },
    ]
  },
  squat_wide_stance_barbell: {
    key: "squat_wide_stance_barbell",
    displayName: "Wide-Stance Squat (Barbell)",
    measures: [
      { key: "lb", displayName: "lb", unit: ExerciseUnit.lb },
      { key: "reps", displayName: "reps", unit: ExerciseUnit.reps },
    ]
  },
};

export const CIRCUIT_REGISTRY: { [k: string]: Circuit } = {
  hip_thrust_circuit: {
    key: "hip_thrust_circuit",
    exercises: [
      {
        exercise: EXERCISE_REGISTRY["hip_thrust_single_leg"],
        note: null
      },
      {
        exercise: EXERCISE_REGISTRY["hip_thrust"],
        note: null
      },
    ],
  }
}

export const WORKOUT_TEMPLATE_REGISTRY: WorkoutTemplateRegistry = {
  rowing_machine: {
    key: "rowing_machine",
    displayName: "Rowing (Machine)",
    iconKey: "rowing",
    note: "Rowing machine for cardio.",
    drills: [
      {
        exercise: EXERCISE_REGISTRY["rowing_machine"],
        displayName: "Rowing (Machine)",
        iconKey: "",
        note: null,
        sets: [{index: 1, setType: SetType.Warmup}, {index: 1, setType: SetType.Working}],
      },
    ]
  },
  legs: {
    key: "legs",
    displayName: "Legs",
    iconKey: "weight-lifter",
    note: "",
    drills: [
      {
        exercise: EXERCISE_REGISTRY["squat_wide_stance_barbell"],
        displayName: "Wide-Stance Squats (Barbell)",
        iconKey: "",
        note: null,
        sets: [
          {index: 1, setType: SetType.Warmup},
          {index: 2, setType: SetType.Warmup},
          {index: 3, setType: SetType.Warmup},
          {index: 1, setType: SetType.Working},
          {index: 2, setType: SetType.Working},
          {index: 3, setType: SetType.Working},
          {index: 4, setType: SetType.Working},
          {index: 5, setType: SetType.Working}
        ],
      },
      {
        circuit: CIRCUIT_REGISTRY["hip_thrust_circuit"],
        displayName: "Hip Thrust Circuit",
        iconKey: "",
        note: null,
        sets: [
          {index: 1, setType: SetType.Working},
          {index: 2, setType: SetType.Working},
          {index: 3, setType: SetType.Working},
          {index: 4, setType: SetType.Working},
          {index: 5, setType: SetType.Working},
          {index: 6, setType: SetType.Working},
        ],
      },
      {
        exercise: EXERCISE_REGISTRY["squat_split_bulgarian_bodyweight"],
        displayName: "Bulgarian Split Squats (Bodyweight)",
        iconKey: "",
        note: null,
        sets: [
          {index: 1, setType: SetType.Working},
          {index: 2, setType: SetType.Working},
          {index: 3, setType: SetType.Working},
          {index: 4, setType: SetType.Working},
          {index: 5, setType: SetType.Working},
        ],
      },
      {
        exercise: EXERCISE_REGISTRY["hamstring_curl_machine"],
        displayName: "Hamstring Curls (Machine)",
        iconKey: "",
        note: null,
        sets: [
          {index: 1, setType: SetType.Working},
          {index: 2, setType: SetType.Working},
          {index: 3, setType: SetType.Working},
        ],
      }
    ],
  },
};
