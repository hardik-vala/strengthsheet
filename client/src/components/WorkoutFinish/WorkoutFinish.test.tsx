import { render, screen } from "@testing-library/react-native";
import React from "react";
import mockSafeAreaContext from "react-native-safe-area-context/jest/mock";
import { WORKOUT_TEMPLATE_REGISTRY } from "../../data/registry";
import { WorkoutFinish } from "./WorkoutFinish";

jest.mock("react-native-safe-area-context", () => mockSafeAreaContext);

describe("WorkoutFinish", () => {
  it("renders correctly", () => {
    const onDismiss = jest.fn();
    render(
      <WorkoutFinish
        workoutTemplate={WORKOUT_TEMPLATE_REGISTRY["rowing_machine"]}
        onDismiss={onDismiss}
      />
    );

    expect(screen).toMatchSnapshot();
  });
});
