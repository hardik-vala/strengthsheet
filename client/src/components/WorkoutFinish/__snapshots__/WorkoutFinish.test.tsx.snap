// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutFinish renders correctly 1`] = `
<View>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "elevation": 0,
          "paddingHorizontal": 0,
          "paddingTop": 15,
          "zIndex": 0,
        },
        {},
        {},
      ]
    }
    testID="appbar-header-root-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "height": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="appbar-header-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": undefined,
            "paddingHorizontal": 4,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="appbar-header"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "height": 40,
              "margin": 6,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "width": 40,
            }
          }
          testID="icon-button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "flex": 1,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="icon-button-container"
          >
            <View
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityTraits="button"
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              centered={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ],
                ]
              }
              testID="icon-button"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": [
                        {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                  testID="cross-fade-icon-current"
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 24,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 32,
            },
            {
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ],
        ]
      }
    >
      Rowing (Machine) workout complete!
    </Text>
    <View
      color="green"
      style={
        [
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderRadius": 40,
            "height": 80,
            "width": 80,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "marginTop": 5,
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
